internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
```

Comment:
The `AppContext` struct embeds `echo.Context`, which can lead to confusion as it mixes context from the HTTP handler with custom application context. It’s best to avoid embedding frameworks directly unless necessary. Instead, consider adding fields for the values you actually need.